src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,33 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(sacrifice) {
+    if (sacrifice instanceof Herbivore && !sacrifice.hidden) {
+      sacrifice.health -= 50;
+    }
+    Animal.alive = Animal.alive.filter(beastie => beastie.health > 0);
```

Comment:
no needs to filter alive animals each time after a bite. Add a condition like `if (sacrifice <= 0) {}`